lib/ddtrace/transport/http/adapters/unix_socket.rb

```diff
@@ -8,15 +9,14 @@ module HTTP
       module Adapters
         # Adapter for Unix sockets
         class UnixSocket < Adapters::Net
-          DEFAULT_TIMEOUT = 1
-
           attr_reader \
             :filepath,
             :timeout
 
-          def initialize(filepath, options = {})
-            @filepath = filepath
-            @timeout = options.fetch(:timeout, DEFAULT_TIMEOUT)
+          # @deprecated Positional parameters are deprecated. Use named parameters instead.
+          def initialize(filepath = nil, **options)
+            @filepath = filepath || options[:filepath]
+            @timeout = options.fetch(:timeout, Ext::Transport::UnixSocket::DEFAULT_TIMEOUT_SECONDS)
```

Comment:
I'm thinking of removing the default value for time `timeout` and return `nil` instead, unless specified by the user. This will allow the adapter to decide its own default.

I could figure out the timeout on a per-protocol basis in the agent settings resolver, but I think that will leak implementation data into the resolver class.

What do you think of letting the adapter decide the timeout when not specified?

Reply:
Ha, you answered my question in another comment, I'll try to remove it without breaking the current API.

Reply:
> What do you think of letting the adapter decide the timeout when not specified?

Works for me, I don't have strong feelings either way, happy with either option.

Reply:
Done! The correct default timeout will now be filled out by the adapter implementation.